Rewrite osu!mania Ranking Criteria (#2243)
ppy/osu-wiki: wiki/Ranking_Criteria/osu!mania/en.md

# osu!mania

> The **osu!mania Ranking Criteria** is a set of rules and guidelines that apply to the creation of osu!mania-specific difficulties. In order to get an osu!mania-specific difficulty ranked, it is mandatory that the creation obeys to the listed criteria. While **all rules must be followed in any circumstance**, guidelines may be violated under exceptional circumstances. These exceptional circumstances must be warranted by an exhaustive explanation as of why the guideline has been violated and why not violating it will interfere with the overall quality of the creation.

Reviewer: ```suggestion
The **osu!mania Ranking Criteria** are a set of rules and guidelines that apply to the creation of osu!mania-specific difficulties. In order to get an osu!mania-specific difficulty ranked, it is mandatory that the creation obeys to the listed criteria. While **all rules must be followed in any circumstance**, guidelines may be violated under exceptional circumstances. These exceptional circumstances must be warranted by an exhaustive explanation as of why the guideline has been violated and why not violating it will interfere with the overall quality of the creation.
```
same issue on every RC page

Author: applied on all

Rule proposals as well as suggestions are discussed in the [Ranking Criteria Subforum](https://osu.ppy.sh/community/forums/87). Any rule that went through a discussion and community approval process is listed here as it has been agreed on in the respective discussion thread.

## Glossary

> ### Difficulty names

Reviewer: ```suggestion
### Difficulty Names
```

Author: yes

- ![Easy](/wiki/shared/diff/easy-m.png "Easy") **Easy**
- ![Normal](/wiki/shared/diff/normal-m.png "Normal") **Normal**
- ![Hard](/wiki/shared/diff/hard-m.png "Hard") **Hard**
- ![Insane](/wiki/shared/diff/insane-m.png "Insane") **Insane**
- ![Expert](/wiki/shared/diff/expert-m.png "Expert") **Expert**

> 

Reviewer: ```suggestion
```

> Refer to [this thread](https://osu.ppy.sh/forum/t/178700) for alternative difficulty naming methods.

Reviewer: ```suggestion
```

Author: i don't see why this should be deleted?

Reviewer: was already removed in other articles ([osu!](https://github.com/ppy/osu-wiki/pull/2219/files#diff-2d5eb1e30f63183236cf08857f744082L19), [osu!taiko](https://github.com/ppy/osu-wiki/pull/2219/files#diff-178b1a51c2d722219c577b4bc98f0776L16), and osu!catch didn't have it to begin with) from a suggestion by UndeadCapulet on the forums

so this is just to keep it standard across each mode

> ### Common terms

Reviewer: ```suggestion
### Common Terms
```

> - **Column**: Vertical lines on the playfield where notes are placed. The editor uses 0|1|2|3 etc. for timestamp labels. 

Reviewer: apparently the colon is included in the bold in the other RCs... though I feel like how you have it here is better so maybe change that in other RC instead. regex `^(\s*-\s+\*\*.+?):\*\*` => `$1**:`

edit: ok thinking about it again I've actually got no idea, which way is this supposed to go? @Joehuu

> - **Column**: Vertical lines on the playfield where notes are placed. The editor uses 0|1|2|3 etc. for timestamp labels. 

Reviewer: no idea what that second sentence is supposed to mean, "etc." is used without a list..? whole thing can be worded clearer

Author: reworded

> - **Key Mode**: The number of columns present in a difficulty.

Reviewer: ```suggestion
- **Key mode**: The number of columns present in a difficulty.
```

Author: yes

> - **Keysounding**: Hit sounding using samples extremely similar or directly from the song in their respective places. This method of hit sounding usually provides low feedback to the player.

Reviewer: ```suggestion
- **Key sounding**: Hit sounding using samples extremely similar or directly from the song in their respective places. This method of hit sounding usually provides low feedback to the player.
```
think this makes sense considering "hit sounding" is two words. and change term wherever it appears

Author: yes

> - **Slider Velocity**: Speed at which the notes/bar lines move on the playfield. Base slider velocity can be controlled in the timing panel and additional changes can be made through inherited (green) timing points.

Reviewer: ```suggestion
- **Slider velocity**: The speed at which the notes/bar lines move on the playfield. Base slider velocity can be controlled in the timing panel and additional changes can be made through inherited (green) timing points.
```

Author: yes

> - **Long Term Slider Velocity**: A slider velocity change lasting for more than 4 measures.

Reviewer: ```suggestion
- **Long-term slider velocity**: A slider velocity change lasting more than 4 measures.
```
and add dash wherever else this appears

Author: yes

> - **Musical Phrase**: A standalone unit of musical meter that is usually 4 measures long. 

Reviewer: ```suggestion
- **Musical phrase**: A standalone unit of musical meter that is usually 4 measures long. 
```

Author: yes

> - **Short Term Slider Velocity**: A slider velocity change lasting less than 1/2 of a beat.

Reviewer: ```suggestion
- **Short-term slider velocity**: A slider velocity change lasting less than 1/2 of a beat.
```
and add dash wherever else this appears

Author: yes

- **Stutter**: Repeated alternations between two slider velocity values with a noticeable difference in their values.
- **Teleport**: A high slider velocity that causes a note to move at a speed that is hard or impossible to react to.

> ### Pattern Terminology

Reviewer: ```suggestion
### Patterns
```

Author: yes

- **Chordstream (Jumpstream, Handstream, Quadstream)**: Streams that mix different types of chords. 
- **Grace**: Two or more notes pressed in rapid succession, generally using snaps at or above 1/6.
- **Hand**: Three notes pressed at the same time.
- **Inverse**: A type of pattern that replaces all regular notes with long notes in a given section, commonly when there are enough long notes to cover a majority of the playfield.
- **Jack**: Three or more consecutive notes in the same column, usually pressed at a faster pace than surrounding notes.
- **Jump**: Two notes pressed at the same time.
- **Jumptrill**: A trill consisting of two alternating jumps.
- **Minijack**: Two consecutive notes in the same column, usually pressed at a faster pace compared to surrounding notes.
- **Release**: The end of a long note which must be released within a certain timing  window to score the entirety of the note properly.
- **Roll**: A variation of a stream consisting of notes that go from the first to the last column or last to the first column, usually done in quick succession.
- **Shield**: A note that appears in the same column immediately before the release of a long note.

> - **Split jumptrill**: A jumptrill which uses both hands at the same time to perform correctly - i.e two one-handed trills.

Reviewer: ```suggestion
- **Split jumptrill**: A jumptrill which requires the use of both hands at the same time to be performed correctly (i.e two one-handed trills).
```

Author: yes

- **Stream**: Constant notes with the same interval that do not repeat to create a minijack.
- **Trill**: More than two notes that alternate between two columns without interruption.

> Refer to [this thread](https://osu.ppy.sh/help/wiki/Guides/osu!mania_Mapping_Guide) for more pattern terminology.

Reviewer: ```suggestion
Refer to the [osu!mania Mapping Guide](/wiki/Guides/osu!mania_Mapping_Guide) for more pattern terminology.
```

Author: yes

## General

### Rules

> All rules are exactly that: _rules_. They are _not_ guidelines and may _not_ be broken under _any_ circumstances.

Reviewer: ```suggestion
All rules are exactly that: *rules*. They are *not* guidelines and may *not* be broken under *any* circumstances.
```

Author: yes

> - **Osu!Topus (AutoPlay) must be able to achieve the full score (1.000.000) in all difficulties.**

Reviewer: ```suggestion
- **osu!topus! (AutoPlay) must be able to achieve the full score (1,000,000) in all difficulties.**
```

Author: yes

- **Timing changes should have a unified scroll speed by adjusting slider velocity.** This is done by normalizing the scroll speed of all timing sections to the most used one in the beatmap. The number is shown in brackets after the BPM range in the song selection, e.g. BPM: 60-160 (130). Note that this guideline does not affect the ability to add slider velocity changes which are used to create SV effects.
- **Slider velocity changes should follow the general intensity of the song.**
- **Slider velocity changes should be snapped according to the song's beat structure.** If the song is using a straight beat, 1/4, 1/8, 1/16 should be used. If the song is using a swing beat, 1/6 or 1/12 should be used. If the song has a beat in a different position from what is recommended, snapping to an actual beat always takes priority.
- **Avoid spamming short kiai time segments.** Doing so can be distracting and can negatively affect gameplay.
- **Breaks should only be used when a section of music is extremely calm or silent.** Lower difficulties, especially on higher BPM songs, may use breaks to avoid parts of a song that are unreasonable for new players to play.

> ## Difficulty specific

Reviewer: ```suggestion
## Difficulty-specific
```

Author: yes

> Difficulty-specific rules and guidelines do only apply to the difficulty level they are listed for and therefore do not apply to every _osu!mania_ difficulty. Rhythm related guidelines apply to approximately 180 BPM maps. If your song is drastically faster or slower, some variables might be different. Apply reasonable judgement in these cases.

Reviewer: ```suggestion
Difficulty-specific rules and guidelines do only apply to the difficulty level they are listed for and therefore do not apply to every *osu!mania* difficulty. Rhythm related guidelines apply to approximately 180 BPM maps. If your song is drastically faster or slower, some variables might be different. Apply reasonable judgement in these cases.
```

Author: yes

> *Keymode-specific guidelines are constructed based on the most common ranked keymodes (4key and 7key). Any other keymodes should apply these guidelines when applicable*

Reviewer: ```suggestion
*Keymode-specific guidelines are constructed based on the most common Ranked key modes (4 Key and 7 Key). Any other key modes should apply these guidelines when applicable*
```

Author: yes

### Easy

> #### Rules

Reviewer: ```suggestion
#### Rules

```

Author: yes

- **Slider velocity changes that alter the scrolling speed of the beatmap are disallowed.** Slider velocity changes to unify the scroll speed in variable BPM beatmaps are exempt.
- **Scroll normalization must be done in beatmaps with variable BPM.**
- **A long note cannot be released during another long note’s body.** This requires a level of finger independence that players of this difficulty cannot handle.

> #### Guidelines

Reviewer: ```suggestion
#### Guidelines

```

Author: yes

- **Note snappings of consecutive 1/4 and higher should not be used.** Slower rhythms are more appropriate for this difficulty level.
- **Long notes should be held for at least one beat.** Faster release timing is too difficult for players of this level.
- **Objects should not be placed during a 1/1 long note’s hold duration.** However, using an object at the beginning or end of a long note is fine.
- **HP and OD values should not be higher than 7.**
- **Note density should consist of mostly 1/1, occasional 1/2, or slower rhythms.**
- **Avoid using more than 5 consecutive 1/2 notes.**
- **Avoid using chords during consecutive 1/2 note patterns.** However, using a chord to end a pattern is fine.
- **Avoid chords with more than 2 notes.** This includes long note patterns that involve notes in more than two columns.

### Normal

> #### Rules

Reviewer: ```suggestion
#### Rules

```

Author: yes

- **A long note cannot be released during another long note’s body.** This requires a level of finger independence that players of this difficulty cannot handle.

> #### Guidelines

Reviewer: ```suggestion
#### Guidelines

```

Author: yes

- **Slider velocity changes that alter the scrolling speed of the beatmap are disallowed.** Slider velocity changes to unify the scroll speed in variable BPM beatmaps are exempt.
- **Scroll normalization must be done in beatmaps with variable BPM.**

Additional guidelines for *4 key Normal* difficulties: 
- **Avoid chords with more than 2 notes.** This includes long note patterns that involve notes in more than two columns.
- **Avoid anchors consisting of three or more notes.** While they are easy for players to read, longer anchors are strenuous. 

Additional guidelines for *7 key Normal* difficulties: 
- **Avoid chords with more than 3 notes.** This includes long note patterns that involve notes in more than three columns.
Bracket patterns should be used sparingly.
- **Long notes played with adjacent keys that are held simultaneously should be released at the same time.** Long notes with multiple release timings require coordination that players of this level cannot reasonably handle.
- **Avoid anchors consisting of three or more notes.** While they are easy for players to read, longer anchors are strenuous. 

### Hard

> #### Guidelines

Reviewer: ```suggestion
#### Guidelines

```

Author: yes

- **Short term slider velocity changes should be used cautiously.** Changes should only happen for sections of different pacings and the slider velocity should not be drastically variable.

Additional guidelines for *4 key Hard* difficulties: 
- **Avoid chords with more than 3 notes.** This includes long note patterns that involve notes in more than three columns.
- **Avoid anchors consisting of five or more notes.** While they are easy for players to read, longer anchors are strenuous. 
Keep jump distance in jumpstream at a 1/1 or higher interval. Anything denser than that can hurt readability.

Additional guidelines for *7 key Hard* difficulties: 
- **Avoid chords with more than 4 notes.** This includes long note patterns that involve notes in more than four columns.
- **Avoid anchors consisting of five or more notes.** While they are easy for players to read, longer anchors are strenuous. 
- **Keep chord distance in a chordstream at a 1/1 or higher interval.** Anything denser than that can hurt readability.
- **Occasional Triples can be used in a stream with a note that their distances are further than a measure (4 beats).**

### Insane

> #### Guidelines

Reviewer: ```suggestion
#### Guidelines

```

Author: yes

- **Using split jump trills for 2+ beats that leads to a hand is discouraged.** Unless a beatmap uses jacks or other complex patterns frequently, this pattern is difficult to hit. If a hand is necessary, delete the note before it that would create the mini jack.
- **Avoid using more than 9-note split jump trills.** These are more physically exhausting than other patterns of similar density.

Additional guidelines for *7 key Insane* difficulties: 
- **Avoid using 1/4 jacks with 3+ notes in the middle of chordstream.** These require finger independence that is too difficult for players at this level.
- **In the middle of a long chord stream with a chord every...**
  - ...1/1, there should not be more than 3 notes, except at the beginning/end of a stream, when 4 is acceptable.
  - ...1/2, there should not be more than 2 notes. 
  - ...1/4, it's recommended to use a single note instead unless there is extremely clear musical justification.
- **Bracket streams should be used sparingly.**

### Expert

> #### Guidelines

Reviewer: ```suggestion
#### Guidelines

```

Author: yes

- ![Easy](/wiki/shared/diff/easy-m.png "Easy") **Easy**
- ![Normal](/wiki/shared/diff/normal-m.png "Normal") **Normal**
- ![Hard](/wiki/shared/diff/hard-m.png "Hard") **Hard**
- ![Insane](/wiki/shared/diff/insane-m.png "Insane") **Insane**
- ![Expert](/wiki/shared/diff/expert-m.png "Expert") **Expert**

### Common Terms

- **Column**: Vertical lines on the playfield where notes are placed. The editor uses `0|1|2|3|4|5|6|7|8|9` for timestamp labels. 
- **Key mode**: The number of columns present in a difficulty.
- **Key sounding**: Hit sounding using samples extremely similar or directly from the song in their respective places. This method of hit sounding usually provides low feedback to the player.
- **Slider velocity**: The speed at which the notes/bar lines move on the playfield. Base slider velocity can be controlled in the timing panel and additional changes can be made through inherited (green) timing points.
- **Long-term slider velocity**: A slider velocity change lasting more than 4 measures.
- **Musical phrase**: A standalone unit of musical meter that is usually 4 measures long. 
- **Short-term slider velocity**: A slider velocity change lasting less than 1/2 of a beat.

> - **Stutter**: Repeated alternations between two slider velocity values with a noticeable difference in their values.

Reviewer 2: this term is unused, the definition should be removed?

> - **Teleport**: A high slider velocity that causes a note to move at a speed that is hard or impossible to react to.

Reviewer 2: this term is unused, the definitions should be removed?

### Patterns

- **Anchor**: A pattern that has a relatively high amount of notes in one column compared with another column, usually found within different types of streams.
- **Chord**: Two or more notes pressed at the same time.
- **Chordstream (Jumpstream, Handstream, Quadstream)**: Streams that mix different types of chords. 
- **Grace**: Two or more notes pressed in rapid succession, generally using snaps at or above 1/6.
- **Hand**: Three notes pressed at the same time.

> - **Inverse**: A type of pattern that replaces all regular notes with long notes in a given section, commonly when there are enough long notes to cover a majority of the playfield.

Reviewer 2: this term is unused, the definitions should be removed?

- **Jack**: Three or more consecutive notes in the same column, usually pressed at a faster pace than surrounding notes.
- **Jump**: Two notes pressed at the same time.

> - **Jumptrill**: A trill consisting of two alternating jumps.

Reviewer 2: idk if jump trill and jumptrill are two different things but the draft later on mentions jump trill which is confusing since jump and trill are defined so is the thing later a different thing or the same or two things at once

- **Minijack**: Two consecutive notes in the same column, usually pressed at a faster pace compared to surrounding notes.
- **Release**: The end of a long note which must be released within a certain timing  window to score the entirety of the note properly.
- **Roll**: A variation of a stream consisting of notes that go from the first to the last column or last to the first column, usually done in quick succession.
- **Shield**: A note that appears in the same column immediately before the release of a long note.

> - **Split jumptrill**: A jumptrill which requires the use of both hands at the same time to be performed correctly (i.e two one-handed trills).

Reviewer 2: this term is unused, the definitions should be removed?

- **Stream**: Constant notes with the same interval that do not repeat to create a minijack.
- **Trill**: More than two notes that alternate between two columns without interruption.

Refer to the [osu!mania Mapping Guide](https://osu.ppy.sh/help/wiki/Guides/osu!mania_Mapping_Guide) for more pattern terminology.

## General

### Rules

All rules are exactly that: *rules*. They are *not* guidelines and may *not* be broken under *any* circumstances.

> - **osu!topus (AutoPlay) must be able to achieve the full score (1,000,000) in all difficulties.**

Reviewer 2: AutoPlay or autoplay (why the capitalization so weird)

### Guidelines

Guidelines may be broken under exceptional circumstances. These must be warranted by a reasonable explanation as of why the guideline has been broken and why not breaking it will interfere with the overall quality of the creation.

- **Every note should correlate to a sound present in the music.** This should be a distinct sound, but can also coherently represent a continuous sound with an indistinguishable start or end. Exceptions in favor of simplification should be made when following the music completely accurately...
  - would be unreasonable to play with full accuracy.
  - would be effectively represented with a consistent, predictable snap
  - would cause an inappropriate difficulty spike.
- **Avoid using storyboard sampled hitsounds with similar volume to hitsounds on notes.** As storyboarded hitsounds can overwhelm and distort audio feedback from objects in the beatmap, placing hitsounds on a note in the chart should be prioritized. However, storyboarded samples may be used when key sounds are extracted directly from a song and when low difficulties have too few notes to support necessary hitsounds.

> - **Do not use key sounds without hitnormal support.** If hitsounds blend perfectly with the song, feedback for hitting notes is minimal. However, when key sounds extracted directly from a song are used as a replacement for a song’s audio, they do not need hitnormal support.

Reviewer 2: change apostrophe to the `'` that is used everywhere else?

- **Avoid using the same hitsound for multiple notes in a chord.** This causes feedback to be louder than necessary. This also applies to active hitsounds and storyboard hitsounds used at the same timestamp.
- **Timing changes should have a unified scroll speed by adjusting slider velocity.** This is done by normalizing the scroll speed of all timing sections to the most used one in the beatmap. The number is shown in brackets after the BPM range in the song selection, e.g. BPM: 60-160 (130). Note that this guideline does not affect the ability to add slider velocity changes which are used to create SV effects.
- **Slider velocity changes should follow the general intensity of the song.**
- **Slider velocity changes should be snapped according to the song's beat structure.** If the song is using a straight beat, 1/4, 1/8, 1/16 should be used. If the song is using a swing beat, 1/6 or 1/12 should be used. If the song has a beat in a different position from what is recommended, snapping to an actual beat always takes priority.
- **Avoid spamming short kiai time segments.** Doing so can be distracting and can negatively affect gameplay.

> - **Breaks should only be used when a section of music is extremely calm or silent.** Lower difficulties, especially on higher BPM songs, may use breaks to avoid parts of a song that are unreasonable for new players to play.

Reviewer 2: I think BPM is missing from the glossary then on this page?

## Difficulty-specific

Difficulty-specific rules and guidelines do only apply to the difficulty level they are listed for and therefore do not apply to every *osu!mania* difficulty. Rhythm related guidelines apply to approximately 180 BPM maps. If your song is drastically faster or slower, some variables might be different. Apply reasonable judgement in these cases.

*Key mode-specific guidelines are constructed based on the most common Ranked key modes (4 key and 7 key). Any other key modes should apply these guidelines when applicable*

### Easy

#### Rules

- **Slider velocity changes that alter the scrolling speed of the beatmap are disallowed.** Slider velocity changes to unify the scroll speed in variable BPM beatmaps are exempt.
- **Scroll normalization must be done in beatmaps with variable BPM.**

> - **A long note cannot be released during another long note’s body.** This requires a level of finger independence that players of this difficulty cannot handle.

Reviewer 2: `'`?

#### Guidelines

- **Note snappings of consecutive 1/4 and higher should not be used.** Slower rhythms are more appropriate for this difficulty level.
- **Long notes should be held for at least one beat.** Faster release timing is too difficult for players of this level.

> - **Objects should not be placed during a 1/1 long note’s hold duration.** However, using an object at the beginning or end of a long note is fine.

Reviewer 2: gdocs why `'`

> - **HP and OD values should not be higher than 7.**

Reviewer 2: imo this would look neater if HP and OD recommendations were the first guideline in all sections for this stuff

- **Avoid using more than 9 consecutives notes in a trills.** The coordination and hand balance required for this pattern is too difficult for players at this skill level.
- **Long notes should be held for at least 1/4 of a beat.**
- **Objects directly after short-term slider velocity changes should appear at the beatmap’s normal scroll speed.** To do this, short-term slider velocity changes should only be used in empty spaces between notes and should equalize to a distance of 1.0x.
- **Long-term slider velocity changes should be between 0.8x and 1.10x.**
- **Slider Velocity gimmicks should be reactable within 1/2 of a beat or more.**

If Hard is the *lowest difficulty of a beatmap set*, it has to abide by these additional **guidelines**:
- **Chords within streams are discouraged.** The density of these patterns is often too difficult for players of this skill level, especially when used on the lowest difficulty of a beatmap set. However, using a chord at the beginning/end of a stream is fine.
- **Short-term slider velocity changes should be used cautiously.** Changes should only happen for sections of different pacings and the slider velocity should not be drastically variable.

> Additional guidelines for *4 key Hard* difficulties: 

Reviewer 2: upper half of this draft calls it 4-Key and this calls it 4 key, decide for one way?
